### PR TITLE
Tasks list redesign

### PR DESCRIPTION
**Pull request by Innders**
## Changelog Descript

- Redesigned the task list item to be cleaner, clearer and more compact.

tasks_list_view_redesign_v001

## Additional Information

The goal was to make the list item more consistent and concise. Only essential information is show at the start, status, folder, task. The project code has been moved away and the status background removed. The width of the folder and task columns have been fixed, so it's easier to scan the list quickly.

The path is shown when hovering to allow for extra context without being overloaded with information when not hovering.

The right columns can be more dynamic as they are secondary information and less visually distracting.

image

### CODE REVIEW

**Comment by martastain** [comment]
I assume it is not a normal use case, but the list is extremely slow (basically frozen) with 100s of tasks. kanban view actually performs MUCH better. maybe there could be a hard limit for tasks and just don't show more than X?

**Comment by BigRoy** [comment]
> I assume it is not a normal use case, but the list is extremely slow (basically frozen) with 100s of tasks

Just wanted to mention that in a simple test case on my own machine it was quite trivial to get myself up to say 100+ tasks - not 1000s, but 100 isn't that crazy amount. If you're scheduled for a project over the next few months that has 100s of shots you can hit those numbers easily, especially if you're lightly involved with e.g. lighting and the comp and are maybe assigned to both.

Similar when responsible for both the modeling and lookdev for many small assets.

**Comment by Innders** [comment]
> list is extremely slow (basically frozen) with 100s of tasks.

I doesn't freeze for me but maybe that's because I've now merged the shortcuts improvements from develop. Before it was re-rendering every single time your mouse moved...

But it is still quite choppy, I will look at ways to improve it. Kanban does lazy loading and if the task isn't in view it's dumbed down to just a placeholder div.

File: src/pages/UserDashboardPage/util/getSortedTasks.js [hunk 1]
```diff
@@ -5,11 +5,23 @@ export const getSortedTasks = (tasks = [], sortBy = []) => {
   return [...tasks].sort((a, b) => {
     for (let i = 0; i < sortBy.length; i++) {
       const { id, sortOrder } = sortBy[i]
-      if (a[id] < b[id]) {
-        return sortOrder ? -1 : 1
-      }
-      if (a[id] > b[id]) {
-        return sortOrder ? 1 : -1
+      if (id.toLowerCase().includes('date')) {
```
[the remark this answers is not in the document]
Comment (Innders) [reply src/pages/UserDashboardPage/util/getSortedTasks.js:8]: A lot easier to read, thanks.

File: src/pages/UserDashboardPage/UserDashboardTasks/UserDashboardList/UserDashboardList.jsx [hunk 2]
```diff
@@ -31,8 +33,24 @@ const UserDashboardList = ({
       // sort by id
       return [...groupedFields].sort((a, b) => {
         if (asc) {
+          // If one group has tasks and the other does not, put the group without tasks at the end
```
[the remark this answers is not in the document]
Comment (Innders) [reply src/pages/UserDashboardPage/UserDashboardTasks/UserDashboardList/UserDashboardList.jsx:36]: I like it! I've just adjusted the variable names to make it even clearer.

**Pull request merged**